clients/rospy/src/rospy/impl/tcpros_service.py

```diff
@@ -137,6 +136,7 @@ def contact_service(resolved_name, timeout=10.0):
                 if first:
                     first = False
                     rospy.core.logerr("wait_for_service(%s): failed to contact [%s], will keep trying"%(resolved_name, uri))
+            time.sleep(0.3)
```

Comment:
The ``rosgraph.MasterException`` from the nested ``contact_service`` function.
The problem is the immediately re-trying of the call. This spams the ros master with service lookups without any sleep.

For me the original logic seems:
* If can contact the service, return immediately. Else sleep.
* Ohh, we should add some exception checking somewhere here